pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -121,7 +123,7 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
       return;
     }
 
-    Collection<FieldSpec> fieldSpecs = schema.getAllFieldSpecs();
+    Set<FieldSpec> fieldSpecs = new HashSet<>(schema.getAllFieldSpecs());
```

Comment:
Can you make this a `LinkedHashSet` to maintain deterministic iteration order? I have a branch this would break.